Automattic/wp-calypso
# state: migrate viewers to redux

#### Changes proposed in this Pull Request

* Remove `Viewers` flux store and related actions
* Remove `ViewersData` component and create a `QueryViewers` component
* Create reducer and data layer handlers for viewers data
* Retrieve viewers data from redux store

#### Testing instructions

Note: Viewers are only visible in the UI for pages which are private. You can access the viewers tab, however, by going to the `/people/viewers/:site` route even it it's not private.  Depending on what sites you have available it may make sense to look for a site with many viewers. 

1. Open a site with viewers on `/people/viewers/:site`
2. Make sure data shows up. Scroll down and look closely at the Network tab in your browser devtools. Is paginated data being fetched correctly?
3. Switch your active site. Are the page params reset correctly when fetching viewers for the site you switched to?
4. Now go back or try removing a viewer for the currently active site. Is the network request successful and the user removed from the list?

Fixes #24129
